### PR TITLE
issue-586 allow migration from none to forward

### LINKED ISSUE #586: Allow promotion of compatibility from none
Currently an event type that has compatibility mode "none" cannot be promoted to another compatibility mode. Since the other modes are stricter it should be possible to promote an event type to both "forward" and "compatible" without any impact on consumers of the events.

### PR DESCRIPTION
**Pull request by rcillo**
Fix #586

### CODE REVIEW

**Comment by adyach** [comment]
@rcillo Isn't it necessary to validate compatibility mode migration?

**Comment by rcillo** [comment]
@adyach could you elaborate on this?

**Comment by adyach** [comment]
discussed it offline with @rcillo, it looks fine 👍

**Comment by adyach** [comment]
👍

File: src/main/java/org/zalando/nakadi/validation/schema/CompatibilityModeChangeConstraint.java [hunk 1]
```diff
@@ -14,7 +14,7 @@
     final Map<CompatibilityMode, List<CompatibilityMode>> allowedChanges = ImmutableMap.of(
             CompatibilityMode.COMPATIBLE, Lists.newArrayList(CompatibilityMode.COMPATIBLE),
             CompatibilityMode.FORWARD, Lists.newArrayList(CompatibilityMode.FORWARD, CompatibilityMode.COMPATIBLE),
-            CompatibilityMode.NONE, Lists.newArrayList(CompatibilityMode.NONE)
+            CompatibilityMode.NONE, Lists.newArrayList(CompatibilityMode.NONE, CompatibilityMode.FORWARD)
```
Comment (musiKk) [inline src/main/java/org/zalando/nakadi/validation/schema/CompatibilityModeChangeConstraint.java]: I'm not intimate with the code base so forgive me for asking but does this include the migration from "none" to "compatible" which should also be possible in my opinion?

Comment (lmontrieux) [inline src/main/java/org/zalando/nakadi/validation/schema/CompatibilityModeChangeConstraint.java]: @musiKk for that, you'll have to migrate from `none` to `forward` first, then from `forward` to `compatible`.

Comment (rcillo) [inline src/main/java/org/zalando/nakadi/validation/schema/CompatibilityModeChangeConstraint.java]: @musiKk we expect users to go step by step and not jump directly.

Comment (rcillo) [inline src/main/java/org/zalando/nakadi/validation/schema/CompatibilityModeChangeConstraint.java]: Also, implementing a direct upgrade, skipping forward, would required a lot more changes.

Comment (musiKk) [inline src/main/java/org/zalando/nakadi/validation/schema/CompatibilityModeChangeConstraint.java]: Ok, wanted to make sure it's a conscious decision.

The background for my question was: For us schema evolution is rather slow as we automatically register/update event types on service deployments. In this mode requiring an extra evolution means an extra release.

Comment (rcillo) [inline src/main/java/org/zalando/nakadi/validation/schema/CompatibilityModeChangeConstraint.java]: 👍 good to know how users integrate. We can keep this direct upgrade as a feature request.

**Comment by rcillo** [comment]
@adyach could you approve again. I have merged master into this branch.

**Comment by adyach** [comment]
👍

**Pull request merged**